sdk/metrics/src/main/scala/org/typelevel/otel4s/sdk/metrics/data/AggregationTemporality.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.metrics.data
+
+import cats.Hash
+import cats.Show
+
+/** The time period over which the measurements are aggregated.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality]]
+  */
+sealed trait AggregationTemporality
```

Comment:
I kind of hate the name, but I don't have a better one

Reply:
I would prefer a shorter name too. 
But Python, Java, JS, and proto-model use `AggregationTemporality`. I guess we should too :(